### PR TITLE
image/registryclient: set insecure and return early in blobMirroredRepository

### PR DESCRIPTION
**Pull request by soltysh**
This is followup to https://github.com/openshift/library-go/pull/1084 fixing:
- setting `insecure` properly for `blobMirroredRepository`
- return proper error if no alternates found
- unit tests for the above two
- replace `golang.org/x/net/context` with built-in `context` package

### CODE REVIEW

File: pkg/image/registryclient/client_mirrored.go [hunk 1]
```diff
@@ -216,6 +216,9 @@ func (r *blobMirroredRepository) alternates(ctx context.Context, fn func(r Repos
 		if err != nil {
 			return err
 		}
+		if len(alternates) == 0 {
+			return attemptErr
+		}
```
Comment (smarterclayton) [inline pkg/image/registryclient/client_mirrored.go:221]: This looks correct but now I want to see a unit test for it.  This is the real loop we need to have unit tests on anyway.

File: pkg/image/registryclient/client_mirrored.go [hunk 2]
```diff
@@ -216,6 +216,9 @@ func (r *blobMirroredRepository) alternates(ctx context.Context, fn func(r Repos
 		if err != nil {
 			return err
 		}
+		if len(alternates) == 0 {
```
Comment (soltysh) [inline pkg/image/registryclient/client_mirrored.go:219]: @smarterclayton alternatively we could check for nil strategy in https://github.com/openshift/library-go/blob/master/pkg/image/registryclient/client_mirrored.go#L207 similarly to [initialRepos](https://github.com/openshift/library-go/blob/master/pkg/image/registryclient/client_mirrored.go#L122) or [errorRepos](https://github.com/openshift/library-go/blob/master/pkg/image/registryclient/client_mirrored.go#L145)

**Review by ricardomaraschini — APPROVED** [review]
/lgtm

**Pull request merged**